skyvern/webeye/browser_factory.py

```diff
@@ -66,40 +67,37 @@ async def listen_to_download(download: Download) -> None:
         try:
             workflow_run_id = kwargs.get("workflow_run_id")
             task_id = kwargs.get("task_id")
+            download_dir = Path(get_download_dir(workflow_run_id=workflow_run_id, task_id=task_id))
 
             async with asyncio.timeout(BROWSER_DOWNLOAD_TIMEOUT):
                 file_path = await download.path()
+                suffix = None
+                download_url_suffix = Path(download.url).suffix
+                suggested_filename_suffix = Path(download.suggested_filename).suffix
                 if file_path.suffix:
-                    return
-
-                LOG.info(
-                    "No file extensions, going to add file extension automatically",
-                    workflow_run_id=workflow_run_id,
-                    task_id=task_id,
-                    suggested_filename=download.suggested_filename,
-                    url=download.url,
-                )
-                suffix = Path(download.suggested_filename).suffix
-                if suffix:
+                    suffix = file_path.suffix
+                elif suggested_filename_suffix:
                     LOG.info(
                         "Add extension according to suggested filename",
                         workflow_run_id=workflow_run_id,
                         task_id=task_id,
-                        filepath=str(file_path) + suffix,
+                        filepath=str(file_path) + suggested_filename_suffix,
                     )
-                    file_path.rename(str(file_path) + suffix)
-                    return
-                suffix = Path(download.url).suffix
-                if suffix:
+                    suffix = suggested_filename_suffix
+                elif download_url_suffix:
                     LOG.info(
                         "Add extension according to download url",
                         workflow_run_id=workflow_run_id,
                         task_id=task_id,
-                        filepath=str(file_path) + suffix,
+                        filepath=str(file_path) + download_url_suffix,
                     )
-                    file_path.rename(str(file_path) + suffix)
-                    return
-                # TODO: maybe should try to parse it from URL response
+                    suffix = download_url_suffix
+
+                name = Path(file_path).name
+                if suffix:
+                    name += suffix
+                new_file_path = download_dir / name
+                await download.save_as(new_file_path)
```

Comment:
what's the specific bug in the original logic?

Reply:
because of this error downloads were not saved @LawyZheng , as `suffix` was `None` @LawyZheng

Reply:
as long as we set the download_paths in the browser args. i don't think we need this change. 
https://github.com/Skyvern-AI/skyvern/blob/583075899fc9938d63ca37f6c1005e3b7306de4a/skyvern/webeye/browser_factory.py#L294

https://github.com/Skyvern-AI/skyvern/blob/583075899fc9938d63ca37f6c1005e3b7306de4a/skyvern/webeye/browser_factory.py#L316